### DISCUSSION TITLE
How to register serialisation class on server

### DISCUSSION THREAD

**Discussion by abhishek852008** [hazelcast/hazelcast-cpp-client#826] [Q&A] [answered]
In IdentifiedDataSerializer example only client point of view is shown, i have also implemented it. But struggling to find anywhere, how to register my class on server side in c++.

**Comment by ihsandemir** [comment, marked as answer]
Hello @abhishek852008 Sorry for late response, I missed your question. Here is the answer:

For the server side, you have to have a Java code implemented and this class jar needs to be put on the server's classpath so that the server also sees the same class. The Java side class implementation details can be found at [this](https://docs.hazelcast.com/imdg/4.2/serialization/implementing-dataserializable.html#identifieddataserializable) page. You have to make sure that the class and factory id should be exactly the same as in C++ client implementation. Here is the [code sample](https://github.com/hazelcast/hazelcast-code-samples/tree/master/serialization/identified-data-serializable) for the Java side implementation.
